src/main/java/org/opentripplanner/middleware/tripMonitor/jobs/CheckMonitoredTrip.java

```diff
@@ -371,41 +499,29 @@ public boolean shouldSkipMonitoredTripCheck() throws Exception {
                 }
             }
 
-            // calculate the next possible itinerary. If the calculation failed, skip this trip check.
-            if (!calculateNextItinerary()) return true;
-
-            // check if the matching itinerary has already ended. This could occur on the very first run of
-            // CheckMonitoredTrip for itineraries that occur today, but has already ended. Up until this point, the
-            // matchingItinerary wasn't calculated, so it wasn't known when the end time was.
-            if (matchingItinerary.endTime.before(DateTimeUtils.nowAsDate())) {
-                // itinerary is done today, advance to the next day and recheck
+            // Check if the CheckMonitoredTrip is being ran for the first time for this trip and if the trip's saved
+            // itinerary has already ended for the day. In that case, advance until the next day.
+            if (previousMatchingItinerary == null && trip.itinerary.endTime.after(DateTimeUtils.nowAsDate())) {
```

Comment:
It might be helpful to add some convenience methods to Itinerary to help with the readability of some of these endTime checks. Instead of

```
trip.itinerary.endTime.after(DateTimeUtils.nowAsDate())
```
We could just have:
```
trip.itinerary.hasEnded()
```

There are a few other places where endTime is checked. And there's at least one place (`updateTripStatus`) where a check like `trip.itinerary.inProgress()` could be useful.

Reply:
Refactored in 1987213b6754ffa98f4748820b628be64f9443f4